flik/linesearch/test/test_conditions.py

```diff
@@ -1 +1,31 @@
 """Tests for flik.linesearch.conditions."""
+import numpy as np
+from numpy.testing import assert_raises
+from conditions import soft_wolfe
+
+def test_soft_wolfe():
+    """ Test soft wolfe/curvature condition."""
+    #set up parameters
+    def gradf(val):
+        return 2*val
+    direction = np.array([2., 1., 3.])
+    alpha = 0.1
+    valf = np.array([2., 1., 3.])
+
+    # Check input quality
+    valf_w = [1, 2, 3]
+    assert_raises(TypeError, soft_wolfe, gradf, valf_w, alpha, direction)
+    gradf_w = 2
+    assert_raises(TypeError, soft_wolfe, gradf_w, valf, alpha, direction)
+    alpha_w = 1
+    assert_raises(TypeError, soft_wolfe, gradf, valf, alpha_w, direction)
+    direction_w = [2., 1., 3.]
+    assert_raises(TypeError, soft_wolfe, gradf, valf, alpha, direction_w)
+    direction_w = np.array([2., 1.])
+    assert_raises(TypeError, soft_wolfe, gradf, valf, alpha, direction_w)
+    const2_w = 1.1
+    assert_raises(TypeError, soft_wolfe, gradf, valf, alpha, direction, const2_w)
+
+    # Check condition
+    assert soft_wolfe(gradf, valf, alpha, direction)
+    assert soft_wolfe(gradf, valf, alpha, -direction) is False
```

Comment:
use `not soft_wolfe(gradf, valf, alpha, -direction)` instead of `soft_wolfe(gradf, valf, alpha, -direction) is False`